src/SessionManager.h

```diff
@@ -0,0 +1,242 @@
+
+	void Weird(const char* name, const Packet* pkt,
+	           const char* addl = "", const char* source = "");
+	void Weird(const char* name, const IP_Hdr* ip,
+	           const char* addl = "");
+
+	[[deprecated("Remove in v5.1. Use packet_mgr->GetPacketFilter().")]]
+	detail::PacketFilter* GetPacketFilter(bool init=true);
+
+	unsigned int CurrentSessions()
+		{
+		return session_map.size();
+		}
+
+	[[deprecated("Remove in v5.1. Use CurrentSessions().")]]
+	unsigned int CurrentConnections() { return CurrentSessions(); }
+
+	/**
+	 * Main entry point for processing packets destined for session analyzers. This
+	 * method is called by the packet analysis manager when after it has processed
+	 * an IP-based packet, and shouldn't be called directly from other places.
+	 *
+	 * @param t The timestamp for this packet.
+	 * @param pkt The packet being processed.
+	 * @param len The number of bytes that haven't been processed yet by packet
+	 * analysis.
+	 */
+	void ProcessTransportLayer(double t, const Packet *pkt, size_t len);
+
+	/**
+	 * Returns a wrapper IP_Hdr object if \a pkt appears to be a valid IPv4
+	 * or IPv6 header based on whether it's long enough to contain such a header,
+	 * if version given in the header matches the proto argument, and also checks
+	 * that the payload length field of that header matches the actual
+	 * length of \a pkt given by \a caplen.
+	 *
+	 * @param caplen The length of \a pkt in bytes.
+	 * @param pkt The inner IP packet data.
+	 * @param proto Either IPPROTO_IPV6 or IPPROTO_IPV4 to indicate which IP
+	 *        protocol \a pkt corresponds to.
+	 * @param inner The inner IP packet wrapper pointer to be allocated/assigned
+	 *        if \a pkt looks like a valid IP packet or at least long enough
+	 *        to hold an IP header.
+	 * @return 0 If the inner IP packet appeared valid, else -1 if \a caplen
+	 *         is greater than the supposed IP packet's payload length field, -2
+	 *         if the version of the inner header does not match proto or
+	 *         1 if \a caplen is less than the supposed packet's payload length.
+	 *         In the -1 case, \a inner may still be non-null if \a caplen was
+	 *         long enough to be an IP header, and \a inner is always non-null
+	 *         for other return values.
+	 */
+	int ParseIPPacket(int caplen, const u_char* const pkt, int proto,
+	                  IP_Hdr*& inner);
+
+	unsigned int SessionMemoryUsage();
+	unsigned int SessionMemoryUsageVals();
+
+	[[deprecated("Remove in v5.1. Use SessionMemoryUsage().")]]
+	unsigned int ConnectionMemoryUsage() { return SessionMemoryUsage(); }
+	[[deprecated("Remove in v5.1. Use SessionMemoryUsageVals().")]]
+	unsigned int ConnectionMemoryUsageConnVals() { return SessionMemoryUsageVals(); }
+
+	unsigned int MemoryAllocation();
+
+	// TODO: should this move somewhere else?
+	analyzer::tcp::TCPStateStats tcp_stats;	// keeps statistics on TCP states
+
+private:
+
+	class StatBlocks {
```

Comment:
Could this move into the implementation file? Seems quite distracting to have all that code here. We can just store a pointer to an instance inside this class I'd say.

Reply:
Yep, moved.